### ISSUE TITLE
Undefined tool name in StackTrace view.

### ISSUE THREAD

**Issue by sangee2004** [gptscript-ai/gptscript#475]
Steps to reproduce the problem:
1. Launch UI using - ` GPTSCRIPT_CHAT_UI_TOOL=github.com/gptscript-ai/ui@v2 gptscript --ui `

2. Execute the following script that uses credentials:

```
chat: true
description: This a search chatbot.
tools: github.com/gptscript-ai/search/duckduckgo,  sys.chat.finish

This is a chatbot. Finish the chat when users type bye or quit.
```

3. Chat with the script to get some information from web. 
4. Use "View Stack Trace" option .

<img width="927" alt="Screenshot 2024-06-10 at 3 55 37 PM" src="https://github.com/gptscript-ai/gptscript/assets/122560863/7351e2e7-189e-4815-b98f-654dd3b489ff">

Notice that the tool name appears as "undefined" in stack trace text.

**Comment by njhale** [comment]
@sangee2004 not getting this issue anymore with main of the UI tool.

Could you try again to see if you're still encountering it?

**Comment by tylerslaton** [comment]
I fixed this in a commit awhile ago but it looks like it slipped through the cracks of me updating issues. Thanks @njhale. Nick's correct @sangee2004 you should no longer see this.

**Comment by sangee2004** [comment]
@tylerslaton @njhale I tested with `gptscript version v0.9.2+0f32a9d8` 

I dont see `Loaded Undefined` anymore instead I see `Ran [object Object]` in `Calls` section. which also is not providing any info to the user.

We expect to see tool name instead ? We have the tool name in `Messages` section already.



![Image](https://github.com/gptscript-ai/gptscript/assets/122560863/1d47eee1-e7ad-4d32-982d-c95fad63a88a)

**Comment by njhale** [comment]
@tylerslaton is that [object Object] supposed to be a tool name?

**Comment by tylerslaton** [comment]
> @tylerslaton is that [object Object] supposed to be a tool name?

Yep, looks like it's an edge case I didn't catch. We can keep this open to track that.
